### PR TITLE
Proposed solution

### PR DESCRIPTION
**Pull request by bastbijl**

### CODE REVIEW

File: src/WeaverRelation.coffee [hunk 1]
```diff
@@ -14,12 +15,14 @@ class WeaverRelation
     Promise.resolve([])
 
   add: (node) ->
-    @nodes[node.id()] = node
-    @pendingWrites.push(Operation.Node(@parent).createRelation(@key, node.id()))
+    nodeId = node.id() if node.id
```
Comment (malamili) [inline src/WeaverRelation.coffee]: There is a better way in coffeescript
`nodeId = if node.id then node.id() else node`

File: src/WeaverRelation.coffee [hunk 2]
```diff
@@ -14,12 +15,14 @@ class WeaverRelation
...
+    nodeId = node if not node.id
+    @nodes[nodeId] = node if node.id
```
Comment (malamili) [inline src/WeaverRelation.coffee]: This is going to break, apparently this function NEEDS a node always, right?

File: src/WeaverRelation.coffee [hunk 3]
```diff
@@ -14,12 +15,14 @@ class WeaverRelation
...
+    @pendingWrites.push(Operation.Node(@parent).createRelation(@key, nodeId))
 
   remove: (node) ->
-    delete @nodes[node.id()]
-    @pendingWrites.push(Operation.Node(@parent).removeRelation(@key, node.id()))
+    delete @nodes[node.id() or node]
```
Comment (malamili) [inline src/WeaverRelation.coffee]: I think this throws an error if node is a string

File: test/WeaverNode.test.coffee [hunk 4]
```diff
@@ -89,6 +89,17 @@ describe 'WeaverNode test', ->
       assert.isDefined(loadedNode.relation('comesBefore').nodes[bar.id()])
     )
 
+  it 'should add a new relation with static reference', ->
+    foo = new Weaver.Node()
+    bar = new Weaver.Node()
+    foo.relation('comesBefore').add(bar.id())
```
Comment (malamili) [inline test/WeaverNode.test.coffee]: its not testing .add(node) with node a string
its not testing removing a node when node is a string

both will fail i think

**Comment by malamili** [comment]
In general, I think these features are not worth the extra effort, possible extra fail paths, this overhead time of reviewing, documenting, testing etc..

I suggest next time to discuss these features on the JIRA board first before implementing, how small they may seem.

File: test/WeaverNode.test.coffee (hunk 4 above)
Comment (bastbijl) [inline test/WeaverNode.test.coffee]: it is testing the add with a string
remove test i will add

Comment (malamili) [inline test/WeaverNode.test.coffee]: oh right, i see

**Pull request closed without merging**